auth/oidc.go

```diff
@@ -245,21 +249,26 @@ func (opm OIDCProviderMap) Stop() {
 func (op *OIDCProvider) GetClient(buildCallbackURLFunc OIDCCallbackURLFunc) *OIDCClient {
 	// Initialize the client on first request. If the callback URL isn't defined for the provider,
 	// uses buildCallbackURLFunc to construct (based on current request)
-	op.clientOnce.Do(func() {
-		var err error
-		// If the redirect URL is not defined for the provider generate it from the
-		// handler request and set it on the provider
-		if op.CallbackURL == nil || *op.CallbackURL == "" {
-			callbackURL := buildCallbackURLFunc(op.Name, op.IsDefault)
-			if callbackURL != "" {
-				op.CallbackURL = &callbackURL
+	if !op.clientInit.IsTrue() { // If client not initialized
+		op.clientInitLock.Lock()
+		if !op.clientInit.IsTrue() {
+			var err error
+			// If the redirect URL is not defined for the provider generate it from the
+			// handler request and set it on the provider
+			if op.CallbackURL == nil || *op.CallbackURL == "" {
+				callbackURL := buildCallbackURLFunc(op.Name, op.IsDefault)
+				if callbackURL != "" {
+					op.CallbackURL = &callbackURL
+				}
+			}
+			if err = op.initOIDCClient(); err != nil {
+				base.Errorf("Unable to initialize OIDC client: %v", err)
```

Comment:
As discussed, let's change GetClient to return (*OIDCClient, error), and have callers check for error instead of nil client.  With that in place can remove the Errorf logging here and rely on the callers to log appropriately (review existing callers to include the err when logging).